src/store/ChartStore.js

```diff
@@ -17,6 +18,7 @@ class ChartStore {
     constructor(mainStore) {
         this.id = ++ChartStore._id_counter;
         this.mainStore = mainStore;
+        this.routingStore = new RoutingStore();
```

Comment:
@shayanbinary should be `this.routingStore = this.mainStore.routing` or even better

```javascript
get routingStore() {
    return this.mainStore.routing;
}
```